src/grid.h

```diff
@@ -18,7 +18,7 @@ struct Grid {
     bool tinyborder;
     bool folded {false};
 
-    Cell *&C(int x, int y) const {

```

Comment:
You probably want to make a const and non-const version of this function, such that you don't have to make functions like `Save` and others non-const for no reason

Reply:
Could you explain to me how I can achieve this? I tried to add a second function, but the compiler only throws errors. I suspect that before this commit, cells is just a pointer to some data structure outside of `Grid`. After this commit, `cells` as a vector lives within the data structure and having a reference to the `Cell*` items, the vector itself is modifiable and cannot be marked as const?

Reply:
By also adding a `Cell *C(int x, int y) const {` (note the absence of `&`).
You then need to check that callers of `C()` correctly receive the return value in a `Cell *&` or `auto &` only when they need to modify and the compiler should select the correct one.
`Save()` for example has no need to modify anything.

Reply:
Ok, solved this by defining an extra macro because auto deduction does not work in the way I wanted.